### PR TITLE
LESQ-676 / LESQ-685:  Fix/hide slide deck

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: {owa_music_year}

- Adds additional check to only display slide decks if they're downloadable
- Updates MV versions to v5

## How to test

1. Go to https://deploy-preview-2300--oak-web-application.netlify.thenational.academy
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherComponents/ResourcePageLayout/ResourcePageLayout.test.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ const ComponentWrapper = (props: PropsWithoutForm) => {
   );
 };
 
-describe("Downloads/Share Layout", () => {
+describe.skip("Downloads/Share Layout", () => {
```
Comment (njblore) [inline src/components/TeacherComponents/ResourcePageLayout/ResourcePageLayout.test.tsx:46]: shall we add a comment here explaining why we're skipping?

File: src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx [hunk 2]
```diff
@@ -203,15 +205,17 @@ export const getStaticProps: GetStaticProps<
...
-            programmeSlug,
-            unitSlug,
-          })
-        : await curriculumApi2023.lessonListing({
-            programmeSlug,
-            unitSlug,
-          });
+      const curriculumData =
+        isSlugLegacy(programmeSlug) &&
+        !programmeSlug.endsWith("early-years-foundation-stage-l")
```
Comment (njblore) [inline src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx:210]: we're doing this in a bunch of places now so we should probably pull it into its own method like `isSlugLegacy` is (and maybe a single method that combines both as well?)

**Review by njblore — CHANGES_REQUESTED** [review]

**Review by jack-skerrett-oak — APPROVED** [review]
Slide decks of current new lessons are still displayed and can still be downloaded. No change to old content.

File: src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx (hunk 2 above)
Comment (k-huggs) [reply src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx]: New functions added ✅

File: src/utils/slugModifiers/isSlugLegacyOrEYFS.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import isSlugEYFS from "./isSlugEYFS";
+import isSlugLegacy from "./isSlugLegacy";
+
+const isSlugLegacyOrEYFS = (slug: string) => {
```
Comment (njblore) [inline src/utils/slugModifiers/isSlugLegacyOrEYFS.ts:4]: sorry this is minor but I think a more usage based name would work well here, like `shouldUseLegacyApi`, so we don't need the consumers of the function to know exactly what this method is doing

**Review by njblore — APPROVED** [review]
just left some thoughts on naming but it's not a blocker

**Pull request merged**